lib/db/models/Node.ts

```diff
@@ -0,0 +1,35 @@
+import Sequelize from 'sequelize';
+import { db } from '../../types';
+import { Address } from '../../types/p2p';
+
+export default (sequelize: Sequelize.Sequelize, DataTypes: Sequelize.DataTypes) => {
+  const attributes: db.SequelizeAttributes<db.NodeAttributes> = {
+    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
+    nodePubKey: { type: DataTypes.STRING, unique: true },
+    addressesText: { type: Sequelize.TEXT },
+    addresses: {
+      type: Sequelize.VIRTUAL,
+      get(this: db.NodeInstance) {
+        return JSON.parse(this.addressesText);
+      },
+      set(this: db.NodeInstance, value: Address[]) {
+        this.setDataValue('addressesText', JSON.stringify(value));
```

Comment:
I hadn't thought of that but it does make querying for node via address awkward (I think it could still be possible with `SELECT * WHERE addresses LIKE '%host:'1.2.3.4',port:8885%`, but not the nicest query). 

The advantage as I see it is that `addresses` can be updated with a single, simple UPDATE query on handshake and it makes loading nodes from the db a little simpler because it's just querying rows from a single table. But if querying node via address is something we'll do regularly, it should probably be changed. I can't think of a real need to query node by address off the top of my head, can you?

As an aside, updating the addresses on handshake is actually not something that's implemented in this PR. So I will either take care of that here or in a separate PR after this is merged.

Reply:
Main scenario for querying by address would be to block an inbound peer, prior to handshake, based on the node entity (see my other comment). We might end up having other usages as well.

Squeezing a many-to-one relationship in a row is a known problem RDBMS, forcing you to normalize. This is why MongoDB became popular, followed by the JSON columns (PostgreSQL and then MySQL).
Concatenating the array is usually a fine for a dumb column, one that you wouldn't do anything with except projecting. Yes, querying would be possible, but pretty awful. 

At the moment, even if we'll want to query by address, we'll do it in-memory (after deserializing), so no DB support is needed. So I guess you can leave this situation as-is for now, and change it if we'll need to.